default/scripting/buildings/TERRAFORM.focs.txt

```diff
@@ -111,15 +115,15 @@ BuildingType
                     * (100 - 10 * ((PlanetTypeDifference from = Target.PlanetType to = Target.NextBestPlanetType) - 1)) / 100
     location = And [
         Planet
-        Not Contains Building name = "BLD_TERRAFORM"
         OwnedBy empire = Source.Owner
         TargetPopulation low = 1
         Not Planet type = [Asteroids GasGiant]
         Not Planet environment = [Uninhabitable Good]
         Not Contains Building name = "BLD_GAIA_TRANS"
+        Not Enqueued type = Building name = "BLD_GAIA_TRANS"
         Not HasSpecial name = "GAIA_SPECIAL"
-[[FOR_ALL_TERRAFORM_PLANET_TYPES(        Not Contains Building name = "BLD_TERRAFORM_,")]]
-[[FOR_ALL_TERRAFORM_PLANET_TYPES(        Not Enqueued type = Building name = "BLD_TERRAFORM_,")]]
+        [[FOR_ALL_TERRAFORM_PLANET_TYPES(Not Contains Building name = "BLD_TERRAFORM_,")]]
```

Comment:
could you add a comment with an example of what this produces? parsing the script bits within the macro bits is difficult.

Reply:
I meant specifically this macro

Reply:
I didn't even write this, just reformatted it and copied it to Gaia Transformation.
Anyway, I've added a comments.
I also noticed that Gaia Transformation does not for a complex enqueue_location, so I simplified it.